product_pricelist_assortment/models/product_pricelist_assortment_item.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+
+class ProductPricelistAssortmentItem(models.Model):
+    _name = "product.pricelist.assortment.item"
+    _description = "Product Pricelist Assortment Item"
+    _inherit = "product.pricelist.item"
+
+    assortment_filter_id = fields.Many2one(
+        comodel_name="ir.filters",
+        domain=[("is_assortment", "=", True)],
+        string="Assortment",
+        ondelete="restrict",
+        required=True,
+    )
+    pricelist_item_ids = fields.One2many(
+        comodel_name="product.pricelist.item",
+        inverse_name="assortment_item_id",
+        help="Pricelist items created automatically",
+    )
+
+    def _compute_name_and_price(self):
+        super()._compute_name_and_price()
+        for rec in self:
+            if rec.assortment_filter_id:
```

Comment:
I think you can remove line 32 and use just rec.name = rec.assortment_filter_id.name